config/yaml-mp/src/main/java/io/helidon/config/yaml/mp/YamlMpConfigSource.java

```diff
@@ -192,14 +192,26 @@ public static List<ConfigSource> classPath(String resource) {
      * Create from YAML file(s) on classpath with profile support.
      *
      * @param resource resource name to locate on classpath (looks for all instances)
-     * @param profile name of the profile to use
+     * @param profile configuration profile to use, must not be null
      * @return list of config sources discovered (may be zero length)
      */
     public static List<ConfigSource> classPath(String resource, String profile) {
+        return classPath(resource, profile, Thread.currentThread().getContextClassLoader());
```

Comment:
if I remember correctly, `getContextClassLoader()` may return `null`
It would be better to have a private method that accepts class loader and checks for null (and does something reasonable when it is null, such as using classloader of the current class)

Reply:
Added a private method to allow null classloader.